cardano-api/src/Cardano/Api/Address.hs

```diff
@@ -528,6 +536,11 @@ isKeyAddress (AddressInEra (ShelleyAddressInEra _) (ShelleyAddress _ pCred _)) =
     PaymentCredentialByKey _ -> True
     PaymentCredentialByScript _ -> False
 
+shellyAddrToPubKeyHash :: Address ShelleyAddr -> Maybe Plutus.PubKeyHash
```

Comment:
should we put `payment` somehow in the name? to distinguish from the staking public key...